front/src/components/Profile/Profile.tsx

```diff
@@ -0,0 +1,622 @@
+  }>({
+    email: "",
+    mobile: "",
+    user_name: "",
+    display_name: "",
+    role: "",
+    password: "",
+  });
+
+  const [errors, setErrors] = useState<Errors>({});
+  const [user_name, setUserName] = useState(userDetails.user_name);
+  const [display_name, setDisplayName] = useState(userDetails.display_name);
+  const [role, setRole] = useState(userDetails.role);
+  const [email, setEmail] = useState(userDetails.email);
+  const [mobile, setMobile] = useState(userDetails.mobile);
+  const [editedData, setEditedData] = useState<RowData>(InitialRowData);
+  const [rows, setRows] = useState<RowData[]>([]);
+  const [alertShow, setAlertShow] = useState("");
+  const [passwordAlert, setPasswordAlert] = useState("");
+  const [password, setPassword] = useState("");
+  const [currentPassword, setCurrentPassword] = useState("");
+  const [confirmPassword, setConfirmPassword] = useState("");
+  const [newPassword, setNewPassword] = useState<string>("");
+  const [userId, setUserId] = useState<number | undefined>(undefined);
+  const [showPassword1, setShowPassword1] = useState<Boolean>(false);
+  const [showPassword2, setShowPassword2] = useState(false);
+  const [showPassword3, setShowPassword3] = useState(false);
+  const [currentPasswordAlert, setCurrentPasswordAlert] =
+    useState<AlertState | null>(null);
+
+  useEffect(() => {
+    const fetchUserDetails = async () => {
+      try {
+        const session: Session | null = await getSession();
+        if (session) {
+          const { email, mobile, id, role, password } = session.user;
+          const user_name = session.user.userName;
+          const display_name = session.user.displayName;
+          setUserDetails({
+            display_name,
+            user_name,
+            email,
+            mobile,
+            role,
+            password,
+          });
+          setEditedData((prevEditedData) => ({
+            ...prevEditedData,
+            display_name,
+            user_name,
+            email,
+            mobile,
+            id,
+            role,
+            password,
+          }));
+          setUserName(user_name || "");
+          setDisplayName(display_name || "");
+          setEmail(email || "");
+          setMobile(mobile || "");
+          setRole(role || "");
+          setUserId(id);
+          setPassword(password || "");
+        }
+      } catch (error) {
+        console.error("Error fetching user details:", error);
+      }
+    };
+    fetchUserDetails();
+  }, []);
+
+  useEffect(() => {
+    if (userDetails) {
+      setUserName(userDetails.user_name || "");
+      setDisplayName(userDetails.display_name || "");
+      setEmail(userDetails.email || "");
+      setMobile(userDetails.mobile || "");
+      setRole(userDetails.role || "");
+    }
+  }, [userDetails]);
+
+  useEffect(() => {
+    getApplications();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  const getApplications = async () => {
```

Comment:
Give a relevant name for the function

Reply:
relevant name for function has been changed